app/src/main/java/com/picone/go4lunch/presentation/viewModels/RestaurantViewModel.java

```diff
@@ -1,32 +1,211 @@
+                        updateUserForRestaurant(restaurant, restaurantsForName.get(0).getKey());
+                    }
+                });
+    }
+
+    private void addRestaurantWhenNotPersisted(Restaurant restaurant) {
+        restaurant.setRestaurantDailySchedule(new RestaurantDailySchedule(DATE, new ArrayList<>()));
+        addRestaurantInteractor.addRestaurant(restaurant)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new CompletableObserver() {
+                    @Override
+                    public void onSubscribe(Disposable d) {
+
+                    }
+
+                    @Override
+                    public void onComplete() {
+                        Log.i("TAG", "onComplete: restaurant added");
+                        updateUserForRestaurant(restaurant, restaurant.getKey());
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+
+                    }
+                });
+    }
+
+    private void updateUserForRestaurant(Restaurant restaurant, String restaurantKey) {
+        addUserToRestaurantInteractor.addCurrentUserToRestaurant(currentUserMutableLiveData.getValue(), restaurant.getName())
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new CompletableObserver() {
+                    @Override
+                    public void onSubscribe(Disposable d) { }
+                    //Suppress Warning is safe cause current user can't be null.
+                    @SuppressWarnings("ConstantConditions")
+                    @Override
+                    public void onComplete() {
+                        updateUserChosenRestaurantInteractor.updateUserChosenRestaurant
+                                (currentUserMutableLiveData.getValue().getEmail(), new UserDailySchedule(DATE, restaurantKey))
+                                .subscribeOn(Schedulers.io())
+                                .observeOn(AndroidSchedulers.mainThread())
+                                .subscribe(new CompletableObserver() {
+                                    @Override
+                                    public void onSubscribe(Disposable d) { }
+
+                                    @Override
+                                    public void onComplete() {
+                                        Log.i("TAG", "onComplete: updateUserComplete");
+                                    }
+
+                                    @Override
+                                    public void onError(Throwable e) {
+
+                                    }
+                                });
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+
+                    }
+                });
+    }
+
+    private void setDate() {
```

Comment:
today starts as a Date the it is initialized again from a SimpleDateFormat Doesn't make sense

Reply:
ok resolved